flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -224,15 +224,24 @@ private static void validateOptionalPositiveDoubleProperty(Properties config, St
 
 	private static void validateOptionalDateProperty(Properties config, String key, String message) {
 		if (config.containsKey(key)) {
-			try {
-				initTimestampDateFormat.parse(config.getProperty(key));
-				double value = Double.parseDouble(config.getProperty(key));
-				if (value < 0) {
-					throw new NumberFormatException();
+			try{
+				if(!validateDateFormat(config, key)){
+					double value = Double.parseDouble(config.getProperty(key));
+					if (value < 0) {
+						throw new NumberFormatException();
+					}
 				}
-			} catch (ParseException | NumberFormatException e) {
+			} catch(NumberFormatException){
 				throw new IllegalArgumentException(message);
 			}
 		}
 	}
+	public static boolean validateDateFormat(Properties config, String key){
```

Comment:
I don't think it is necessary to have yet another method for this particular case ...
it can be easily written as 2 catch exception branches, which IMO is easier to understand.